src/main/sensors/battery.c

```diff
@@ -174,6 +176,62 @@ void batteryInit(void)
     batteryCriticalVoltage = 0;
 }
 
+uint8_t calculateBatteryPercentageRemaining(void)
+{
+    uint8_t batteryPercentage = 0;
+    if (batteryCellCount > 0) {
+        uint16_t batteryCapacity = batteryConfig()->batteryCapacity;
+
+        if (feature(FEATURE_CURRENT_METER) && batteryCapacity > 0) {
```

Comment:
I think we need an option to force `calculateBatteryPercentageRemaining` to always use battery voltage.
One may fly with batteries of different capacity (or semi-discharged batteries) and would prefer to see the percentage based on voltage - it would be more informative.

Reply:
We can detect when a battery is not fully charged and ignore `batteryConfig()->batteryCapacity`, but detecting an incorrectly configured value would be kinda difficult.

I can see two different valid lines here:

1) Add a CLI setting which lets the user decide to ignore battery capacity during percentage calculation. This makes the code simpler, since all callers for informative purposes (telemetry, OSD, etc...) can use `calculateBatteryPercentageRemaining` while potential future callers (low battery triggered FS?) can still use the lower level APIs directly if the need to.

2) Re-introduce `calculateBatteryCapacityRemainingPercentage` and the previous `calculateBatteryPercentage` but renamed to `calculateBatteryVoltageRemainingPercentage`. Then refactor `calculateBatteryPercentage` to use them. Finally, decide which places should call each function. I think this complicates stuff a bit and makes the reported percentage a bit jumpy when it doesn't need to be, but I like the fact that it doesn't introduce another setting.

Reply:
I'm in favor of CLI option. Let's have two low-level APIs and one high-level API that will chose based on CLI setting. Default for this setting should be `VOLTAGE` to be compatible with current behavior